cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc

```diff
@@ -1807,11 +1807,12 @@ const FunctionDoc is_dst_doc{
      "An error is returned if the values do not have a defined timezone."),
     {"values"}};
 
-const FunctionDoc local_time_doc{
+const FunctionDoc local_timestamp_doc{
     "Convert timestamp to a timezone-naive local time timestamp",
-    ("LocalTime converts a timestamp to a local time of timestamps timezone\n"
-     "and removes timezone metadata. If input is in UTC or doesn't have\n"
-     "timezone metadata, it is returned as is.\n"
+    ("LocalTimestamp converts timezone-aware timestamp to local timestamp\n"
+     "of the given timestamps timezone and removes timezone metadata.\n"
```

Comment:
"timestamps timezone" -> should that be "timestamp's timezone" ? (otherwise I don't fully understand the sentence)

Maybe we can also include "wall clock time" somewhere in the explanation?

Reply:
Changed and added.